FirebaseTestProject/ViewModels/PostData.swift

```diff
@@ -9,10 +9,39 @@ import Foundation
 
 @MainActor class PostData: ObservableObject {
     @Published var posts: [Post] = []
+    var dataType: DataType
     
-    init() {
+    init(dataType: DataType) {
+        self.dataType = dataType
+        setPosts()
+    }
+    
+    func setPosts() {
         Task {
-            await loadPosts()
+            // Favorites HAS to go here for this to work correctly --
+            // the async calls (this one and the posts one) need to happen PRIOR to
+            // setting self.posts
+            let favorites = try await PostService.getFavorites()
+            
+            switch dataType {
+            case .all:
+                await loadPosts()
+            case .singleAuthor(let author):
+                await loadPosts()
+                posts = posts.filter({$0.author == author})
```

Comment:
Can this filtering be done when pulling the data?  Not a problem if it can't be

Reply:
Though the `PostService` might wait to own the filtering

Reply:
Redid this - Completed